### PR TITLE
Refactor integration manager handling into a common place

### PR DESCRIPTION
**Pull request by turt2live**
It was already in a common place, but this is the boilerplate for supporting multiple integration managers, and multiple integration manager sources. 

For https://github.com/vector-im/riot-web/issues/4913 / https://github.com/vector-im/riot-web/issues/10161

### CODE REVIEW

File: src/integrations/IntegrationManagers.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        if (this.hasManager()) {
+            // TODO: TravisR - Handle custom integration managers (widgets)
+            return this._managers[0];
+        } else {
+            return null;
+        }
+    }
+
+    openNoManagerDialog(): void {
+        // TODO: Is it Integrations (plural) or Integration (singular). Singular is easier spoken.
```
Comment (jryans) [inline src/integrations/IntegrationManagers.js]: I feel like we typically say "integration manager", and grammatically I think it makes sense that an "<thing> manager" would manage multiple <thing>s, so let's try to standardise around "integration manager".

Comment (turt2live) [reply src/integrations/IntegrationManagers.js]: I am unbelievably appreciative of this comment. I'm going to go delete a bunch of extraneous s's now.

File: src/integrations/IntegrationManagers.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+limitations under the License.
+*/
+
+import SdkConfig from '../SdkConfig';
+import sdk from "../index";
+import Modal from '../Modal';
+import {IntegrationManagerInstance} from "./IntegrationManagerInstance";
+
+export class IntegrationManagers {
+    static _instance;
```
[the remark this answers is not in the document]
Comment (turt2live) [reply src/integrations/IntegrationManagers.js]: I guess I should test it :|

can we just rewrite it all in a language that doesn't do dumb things?

[the remark this answers is not in the document]
Comment (turt2live) [reply src/integrations/IntegrationManagers.js]: the riot-web layer explodes when it sees the flow imports...

**Pull request merged**